src/win32/tip/tip_edit_session.cc

```diff
@@ -553,23 +554,42 @@ class SyncGetTextEditSessionImpl final
     : public TipComImplements<ITfEditSession> {
  public:
   SyncGetTextEditSessionImpl(wil::com_ptr_nothrow<TipTextService> text_service,
-                             wil::com_ptr_nothrow<ITfRange> range)
-      : text_service_(std::move(text_service)), range_(std::move(range)) {}
+                             wil::com_ptr_nothrow<ITfRange> range,
+                             bool check_composition)
+      : text_service_(std::move(text_service)),
+        range_(std::move(range)),
+        check_composition_(check_composition) {}
 
   // The ITfEditSession interface method.
   // This function is called back by the TSF thread manager when an edit
   // request is granted.
   virtual STDMETHODIMP DoEditSession(TfEditCookie read_cookie) {
-    TipRangeUtil::GetText(range_.get(), read_cookie, &text_);
+    HRESULT result = TipRangeUtil::GetText(range_.get(), read_cookie, &text_);
+    if (FAILED(result)) {
+      return result;
+    }
+    if (check_composition_) {
+      wil::com_ptr_nothrow<ITfContext> context;
+      result = range_->GetContext(&context);
```

Comment:
Is `GetContext` lightweight? If no, how about propagating the context from the caller (i.e. `GetTextSync`)?